observability/test/gke-deployment.yaml

```diff
@@ -0,0 +1,151 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: grpc-server-ns-${JOB_MODE}
```

Comment:
You probably want to put some unique ID in these names so that you can have multiple test runs going in parallel.

Reply:
1. I had removed all of the GKE related code for this iteration for now, because now individual language's interop client/server docker containers are now being run by `docker run` command, things need to be changed on the GKE deployment side. For that change I find it just too much for this PR so I will deal with the GKE aspect of things in a separate PR.  For now, this PR will only execute all the tests in the Kokoro VM.

2. Yes I added a TODO to add a unique identifier to each docker container we just built to make sure each test run is running against that exact docker container just being built.